__tests__/crowdfundings/updatePayment.test.js

```diff
@@ -0,0 +1,154 @@
+  const newPledge = await prepareNewPaidPledge(options)
+  return { ...newPledge }
+}
+
+test('updatePayment: WAITING -> CANCELLED', async (t) => {
+  const { pledgeId } = await prepare()
+  await signIn({ user: Users.Supporter })
+  const status = 'CANCELLED'
+  const { paymentId } = await pgDatabase().public.pledgePayments.findFirst({ pledgeId })
+  const result = await updatePayment({
+    paymentId,
+    status,
+    reason: 'WAITING -> CANCELLED'
+  })
+  t.ok(result.errors, 'invalid transition')
+
+  await signOut()
+  t.end()
+})
+
+test('updatePayment: WAITING -> WAITING_FOR_REFUND', async (t) => {
+  const { pledgeId } = await prepare()
+  await signIn({ user: Users.Supporter })
+  const status = 'WAITING_FOR_REFUND'
+  const { paymentId } = await pgDatabase().public.pledgePayments.findFirst({ pledgeId })
+  const result = await updatePayment({
+    paymentId,
+    status,
+    reason: 'WAITING -> WAITING_FOR_REFUND'
+  })
+  t.ok(result.errors, 'invalid transition')
+
+  await signOut()
+  t.end()
+})
+
+test('updatePayment: WAITING -> REFUNDED', async (t) => {
+  const { pledgeId } = await prepare()
+  await signIn({ user: Users.Supporter })
+  const status = 'WAITING_FOR_REFUND'
+  const { paymentId } = await pgDatabase().public.pledgePayments.findFirst({ pledgeId })
+  const result = await updatePayment({
+    paymentId,
+    status,
+    reason: 'WAITING -> REFUNDED'
+  })
+  t.ok(result.errors, 'invalid transition')
+
+  await signOut()
+  t.end()
+})
+
+test('updatePayment: WAITING -> PAID', async (t) => {
+  const { pledgeId } = await prepare()
+  await signIn({ user: Users.Supporter })
+  const status = 'PAID'
+  const { paymentId } = await pgDatabase().public.pledgePayments.findFirst({ pledgeId })
+  const result = await updatePayment({
+    paymentId,
+    status,
+    reason: 'WAITING -> PAID'
+  })
+  t.notOk(result.errors, 'graphql mutation successful')
+  t.deepEqual(result.data, {
+    updatePayment: { status }
+  }, `payment status is ${status} now and no memberships generated at all`)
+
+  await signOut()
+  t.end()
+})
+
+test('updatePayment: WAITING_FOR_REFUND -> REFUNDED', async (t) => {
+  const { pledgeId } = await prepare()
+  await signIn({ user: Users.Supporter })
+  const { paymentId } = await pgDatabase().public.pledgePayments.findFirst({ pledgeId })
+
+  // WAITING -> PAID
+  await updatePayment({
+    paymentId,
+    status: 'PAID',
+    reason: 'WAITING_FOR_REFUND -> REFUNDED'
+  })
+  // PAID -> WAITING_FOR_REFUND
+  await cancelPledge({ pledgeId })
+
+  // WAITING_FOR_REFUND -> REFUNDED
+  const result = await updatePayment({
+    paymentId,
+    status: 'REFUNDED',
+    reason: 'WAITING_FOR_REFUND -> REFUNDED'
+  })
+  t.notOk(result.errors, 'graphql mutation successful')
+  t.deepEqual(result.data, {
+    updatePayment: { status: 'REFUNDED' }
+  }, `payment status is REFUNDED now and no memberships generated at all`)
+
+  await signOut()
+  t.end()
+})
```

Comment:
please add the same test as before, but after `cancelPledge` instead of `updatePayment` call `modules/crowdfundings/lib/payments/stripe/webhooks/chargeRefunded.js` and see if the `payment.status` switched to `REFUNDED` as well.

Reply:
this is going to be in payPledge as I have the needed context setup there